### PR TITLE
feat: sql merge, ttl merge and deploy index fix

### PR DESCRIPTION
**Pull request by vagetablechicken**
ttl merge in extract index & deploy getnew index
ttl update of table index
multi join keys in sql merge

### CODE REVIEW

File: steps/ut_zookeeper.sh [hunk 1]
```diff
@@ -37,6 +37,11 @@ case $OP in
         echo "Stopping zk ... "
         cd "$THIRDSRC/zookeeper-3.4.14" && ./bin/zkServer.sh stop
         ;;
+    reset)
+        echo "remove zk data and then start zk"
```
Comment (vagetablechicken) [inline steps/ut_zookeeper.sh]: stop->rm->start

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/SqlClusterExecutor.java [hunk 2]
```diff
@@ -387,6 +393,82 @@ public static List<String> validateSQLInRequest(String sql, Map<String, Map<Stri
...
+                uniqueKeysNewName.stream().map(newNamePrefix -> String.format(equalPattern, idx, newNamePrefix, idx))
+                        .collect(Collectors.joining(" and "));
+            }
+        }
+        sb.append(";");
+        return sb.toString();
+    }
+
+    // with table schema, we can except join keys and validate the merged sql
+    public static String mergeSQL(List<String> sqls, List<String> uniqueKey,
```
Comment (vagetablechicken) [inline java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/SqlClusterExecutor.java]: add doc

File: src/base/ddl_parser_test.cc [hunk 3]
```diff
@@ -152,6 +152,67 @@ class DDLParserTest : public ::testing::Test {
...
+    test_same_type(type::TTLType::kAbsoluteTime, {1, 0}, {2, 0}, true, {2, 0});
+    test_same_type(type::TTLType::kAbsoluteTime, {1, 0}, {0, 0}, true, {0, 0});
+    test_same_type(type::TTLType::kAbsoluteTime, {0, 0}, {22, 0}, false, {0, 0});
+    test_same_type(type::TTLType::kAbsAndLat, {10, 20}, {20, 30}, true, {20, 30});
+    test_same_type(type::TTLType::kAbsAndLat, {10, 20}, {5, 30}, true, {10, 30});
+    test_same_type(type::TTLType::kAbsOrLat, {10, 20}, {20, 30}, true, {20, 30});
+    test_same_type(type::TTLType::kAbsOrLat, {10, 20}, {5, 30}, true, {10, 30});
+    // different type
+    // abs + lat
+    test_func(type::TTLType::kAbsoluteTime, {1, 0}, type::TTLType::kLatestTime, {0, 2}, true, type::TTLType::kAbsOrLat,
```
Comment (vagetablechicken) [inline src/base/ddl_parser_test.cc:296]: ut: window unbounded preceding -> abs 0, lat 0?

File: src/base/ddl_parser_test.cc [hunk 4]
```diff
@@ -245,6 +306,22 @@ TEST_F(DDLParserTest, complexJoin) {
...
+        "select * from (select id as merge_id_0, c1 from main) as out0 last join (select id as merge_id_1, sum(c1) "
+        "over w1 from main window w1 as (partition by c1 order by c2 rows between unbounded preceding and current "
+        "row)) as out1 on out0.merge_id_0 = out1.merge_id_1 last join (select id as merge_id_2, t1.c2 from main last "
+        "join t1 on main.c1==t1.c1) as out2 on out0.merge_id_0 = out2.merge_id_2 last join (select id as merge_id_3, "
+        "t1.c2 from main last join t1 order by t1.c2 on main.c1==t1.c1) as out3 on out0.merge_id_0 = out3.merge_id_3 "
+        "last join (select id as merge_id_4, sum(c2) over w1 from main window w1 as (union (select \"\" as id, * from "
+        "t1) partition by c1 order by c2 rows between unbounded preceding and current row)) as out4 on out0.merge_id_0 "
+        "= out4.merge_id_4;";
+    auto index_map = DDLParser::ExtractIndexes(sql, db);
+    LOG(INFO) << index_map;
```
Comment (vagetablechicken) [inline src/base/ddl_parser_test.cc]: add index map check?

File: src/base/ddl_parser.cc [hunk 5]
```diff
@@ -521,7 +527,7 @@ bool IndexMapBuilder::CreateIndex(const std::string& table, const hybridse::node
 
     if (index_map_.find(index) != index_map_.end()) {
         // TODO(hw): ttl merge
```
Comment (vagetablechicken) [inline src/base/ddl_parser.cc]: comment, it'll merge in ToMap

File: src/base/ddl_parser_test.cc (hunk 3 above)
Comment (vagetablechicken) [reply src/base/ddl_parser_test.cc:296]: yep, check by ut DDLParserTest.multiJoin

**Comment by vagetablechicken** [comment]
Test needs update java sdk, openmldb server(nameserver&tablet).
Test way: merge sql, deploy merged sql and check the index of table. No limit, every deployed sql works.

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Review by lqy222 — APPROVED** [review]

**Pull request merged**